packages/ui/src/components/pages/MessageResultPage.tsx

```diff
@@ -170,19 +175,24 @@ export function MessageResultPage({
 
 	const question = thread?.name ?? firstMessage.content?.slice(0, 100);
 	const isFirstMessageSolution = solution && solution.id !== firstMessage.id;
+
+	const nameHtml = useParsedMarkdown(question);
+	const textHtml = useParsedMarkdown(firstMessage.content);
+	const solutionHtml = useParsedMarkdown(solution?.content ?? '');
+
 	const qaHeader: WithContext<QAPage> = {
 		'@context': 'https://schema.org',
 		'@type': 'QAPage',
 		mainEntity: {
 			'@type': 'Question',
-			name: toHTML(question),
-			text: toHTML(firstMessage.content),
+			name: nameHtml,
```

Comment:
I have some concerns about moving the sanitization away from where it's being used - however - maybe it'd be ok if we make this var name  like "safeNameHtml"